### DISCUSSION TITLE
How can I get a new design matrx after running tseriesinterp.m

### DISCUSSION THREAD

**Discussion by HongUKA** [cvnlab/GLMsingle#120] [Q&A] [answered]
after running tseriesinterp.m I've got 4 new timeseries (tcs_interp) for all four conditions each run. But how can I code a new design matrix? with which function or toolbox? 

It is written in GLMSingle Doc that <design> and <data> should have the same number of runs, the same number of time points. But after running tseriesinterp.m the time points of <design> does not match the time poinst of <data> anymore. Is that mean that I should resample my fmri data too? 

thank you in advance for any help!

**Comment by kendrickkay** [comment, marked as answer]
Yes, you should prepare a design matrix that reflects the new sampling rate of your data.

If you did a simple integer upsampling of your data rate, perhaps do something like insert a row(s) of zeros after every row of your original design matrix?  

Like, [1 0 0] becomes   [1 0   0 0    0 0   ]

**Reply by HongUKA** [reply]
I'm sorry, I still don't understand. 

How can I use this result (tcs_interp) from tseriesinterp.m?  Can it be used for slice timing of SPM? The design matrix should be unsampled accordingly e.g. from 100 to 200 rows (stimdur = 1s, TR = 2 s). Then with new data from SPM and new matrix I can call  GLMestimatesingletrial.m?

**Comment by kendrickkay** [comment]
The tcs_interp output presumably is your fMRI data that has been upsampled. (In theory you can also use it to adjust slice timing differences.)

Yes, the design matrix should be upsampled too (though you should not just blindly upsample it using tseriesinterp.m unless you know what you are doing (e.g. use nearest neighbor interpolation)).

Then, with a proper design matrix and a proper upsampled time series, you can call GLMestimatesingletrial.m

**Reply by HongUKA** [reply]
thank you for your explanation!
